diff --git a/vs_toolchain.py b/vs_toolchain.py
--- a/vs_toolchain.py
+++ b/vs_toolchain.py
@@ -69,1 +69,2 @@ MSVS_VERSIONS = collections.OrderedDict([
+    ('18', '18.0'),
     ('2022', '17.0'),  # The VS version in our packaged toolchain.
@@ -77,1 +78,2 @@ MSVC_TOOLSET_VERSION = {
+    '18': 'VC144',
     '2022': 'VC143',
@@ -207,7 +209,7 @@ def GetVisualStudioVersion():
       available_versions.append(version)
       break
     # Detecting VS under possible paths.
-    if version >= '2022':
+    if version not in ('2017', '2019'):
       program_files_path_variable = '%ProgramFiles%'
     else:
       program_files_path_variable = '%ProgramFiles(x86)%'
@@ -238,7 +240,7 @@ def DetectVisualStudioPath():
   # the registry. For details see:
   # https://blogs.msdn.microsoft.com/heaths/2016/09/15/changes-to-visual-studio-15-setup/
   # For now we use a hardcoded default with an environment variable override.
-  if version_as_year >= '2022':
+  if version_as_year not in ('2017', '2019'):
     program_files_path_variable = '%ProgramFiles%'
   else:
     program_files_path_variable = '%ProgramFiles(x86)%'
