diff --git a/config/compiler/BUILD.gn b/config/compiler/BUILD.gn
--- a/config/compiler/BUILD.gn
+++ b/config/compiler/BUILD.gn
@@ -625,12 +625,6 @@ config("compiler") {
       cflags += [ "-ffp-contract=off" ]
     }
 
-    # Enable ELF CREL (see crbug.com/357878242) for all platforms that use ELF
-    # (excluding toolchains that use an older version of LLVM).
-    if (is_linux && !llvm_android_mainline &&
-        default_toolchain != "//build/toolchain/cros:target") {
-      cflags += [ "-Wa,--crel,--allow-experimental-crel" ]
-    }
   }
 
   # C11/C++11 compiler flags setup.
