diff --git a/gni/v8.gni b/gni/v8.gni
--- a/gni/v8.gni
+++ b/gni/v8.gni
@@ -58,6 +58,9 @@ declare_args() {
   # Enable monolithic static library for embedders.
   v8_monolithic = false
 
+  # Hide internal symbols when the monolith is embedded in a shared library.
+  v8_monolithic_hide_symbols = false
+
   # Expose public symbols for native modules of Node.js and Electron. Default
   # is false.
   v8_expose_public_symbols = ""
@@ -313,7 +316,9 @@ if (v8_symbol_level != symbol_level) {
   }
 }
 
-if ((is_posix || is_fuchsia) && (v8_enable_backtrace || v8_monolithic)) {
+if ((is_posix || is_fuchsia) &&
+    (v8_enable_backtrace ||
+     (v8_monolithic && !v8_monolithic_hide_symbols))) {
   v8_remove_configs += [ "//build/config/gcc:symbol_visibility_hidden" ]
   v8_add_configs += [ "//build/config/gcc:symbol_visibility_default" ]
 }
